### PR TITLE
[4.3] Media Manager Ordering

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Base work for the ordering
- Supports: name, size, dimensions, date created and date modified

~The TODO here is to expose to the icons view the ui (ie dropdown, series of buttons, etc) to control the ordering. Right now you have to switch to the table view to select another column for ordering...~ 

### Testing Instructions
Apply the PR and run `npm ci`

Check the grid layout of the media browser and click on the table headers

### Expected result AFTER applying this Pull Request
Screenshot 2023-01-29 at 13 29 48
Screenshot 2023-01-29 at 13 30 11

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
The Upload button is no longer working. Clicking it does nothing.

**Comment by Quy** [comment]
Maybe change it to caret to match elsewhere. Also, the arrow is opposite direction compared to elsewhere.

**Comment by dgrammatiko** [comment]
> Maybe change it to caret to match elsewhere. Also, the arrow is opposite direction compared to elsewhere.

@Quy should be ok now

**Comment by Quy** [comment]
To be consistent elsewhere, the caret direction should reflect the current ordering.

**Comment by dgrammatiko** [comment]
> To be consistent elsewhere, the caret direction should reflect the current ordering.

done

**Comment by Quy** [comment]
Sorry one minor change so the last word is not cut off. Make both dropdowns `col-3`.

39732-dropdown

File: administrator/components/com_media/resources/scripts/store/mutation-types.es6.js [hunk 1]
```diff
@@ -54,3 +54,9 @@ export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY';
 
 // Update item properties
 export const UPDATE_ITEM_PROPERTIES = 'UPDATE_ITEM_PROPERTIES';
+
+// Update srting by
+export const UPDATE_SORT_BY = 'UPDATE_SORT_BY';
+
+// Update srting direction
```
Comment (Quy) [inline administrator/components/com_media/resources/scripts/store/mutation-types.es6.js]:
```suggestion
// Update string by
export const UPDATE_SORT_BY = 'UPDATE_SORT_BY';

// Update string direction
```

Comment (dgrammatiko) [reply administrator/components/com_media/resources/scripts/store/mutation-types.es6.js]: oops it's not string but sorting, It's ok will fix it

File: administrator/components/com_media/tmpl/media/default_texts.php [hunk 2]
```diff
@@ -12,65 +12,78 @@
...
+    'COM_MEDIA_TOGGLE_SELECT_ITEM',
+    'COM_MEDIA_TOOLBAR_LABEL',
+    'COM_MEDIA_UPLOAD_SUCCESS',
+    'ERROR',
+    'JACTION_CREATE',
+    'JAPPLY',
+    'JCANCEL',
+    'JGLOBAL_CONFIRM_DELETE',
+    'JGRID_HEADING_ORDERING_ASC',
+    'JGRID_HEADING_ORDERING_DESC',
```
Comment (Quy) [inline administrator/components/com_media/tmpl/media/default_texts.php]:
```suggestion
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3becfe4830399794baafbb09e4ee70d7e317bcf3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39732">issues.joomla.org/tracker/joomla-cms/39732</a>.</sub>

**Comment by Quy** [comment]
Fix gap.

39741-gap

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 366d1862f2f99e1809bcc994dab4a7e16661fa14<br /><br />THANK YOU!!!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39732">issues.joomla.org/tracker/joomla-cms/39732</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 366d1862f2f99e1809bcc994dab4a7e16661fa14<br /><br />👍<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39732">issues.joomla.org/tracker/joomla-cms/39732</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39732">issues.joomla.org/tracker/joomla-cms/39732</a>.</sub>

**Comment by obuisard** [comment]
Great PR. Thank you Dimitris @dgrammatiko.

**Pull request merged**